integration_tests/follow_me.cpp

```diff
@@ -0,0 +1,148 @@
+#include <iostream>
+#include <cmath>
+#include <thread>
+#include <chrono>
+#include "integration_test_helper.h"
+#include "global_include.h"
+#include "dronecore.h"
+
+using namespace dronecore;
+using namespace std::chrono;
+using namespace std::this_thread;
+
+#ifndef FOLLOW_ME_TESTING
+#define FOLLOW_ME_TESTING
+#endif
+
+void print_flight_mode(Telemetry::FlightMode flight_mode);
+void print_config(FollowMe::Configuration cfg);
+
+TEST_F(SitlTest, FollowMe)
+{
+    DroneCore dc;
+
+    DroneCore::ConnectionResult ret = dc.add_udp_connection();
+    ASSERT_EQ(DroneCore::ConnectionResult::SUCCESS, ret);
+
+    // Wait for device to connect via heartbeat.
+    sleep_for(seconds(2));
+    Device &device = dc.device();
+
+    while (!device.telemetry().health_all_ok()) {
+        std::cout << "waiting for device to be ready" << std::endl;
+        sleep_for(seconds(1));
+    }
+
+    Action::Result action_ret = device.action().arm();
+    ASSERT_EQ(Action::Result::SUCCESS, action_ret);
+
+    device.telemetry().flight_mode_async(
+        std::bind(&print_flight_mode, std::placeholders::_1));
+
+    action_ret = device.action().takeoff();
+    ASSERT_EQ(Action::Result::SUCCESS, action_ret);
+
+    sleep_for(seconds(5));
+
+    ////////////////////////////////////////
+    /// Uses Default PX4 FollowMe config////
+    // Min height: 8mts
+    // Distance to device: 8mts
+    // Responsiveness: 0.5
+    // Device follows from: Behind
+    ////////////////////////////////////////
+    auto defult_cfg = device.followme().get_config();
+    print_config(defult_cfg);
+
+    // start following with default configuration
+    FollowMe::Result followme_result = device.followme().start();
```

Comment:
Why is the API `followme` it should be `follow_me`, that's odd.

Reply:
`followme()` is generated from `classname` which is set in [CMakeLists.txt](https://github.com/dronecore/DroneCore/pull/142/files#diff-1f0169d95c956b412a642e528a116f0fR2). How do I change it ?

Reply:
I see 😄 . I guess I need to look into the cmake magic to fix this.